### PR TITLE
Use generic md api

### PR DESCRIPTION
**Pull request by rubenk**

### CODE REVIEW

**Comment by Habbie** [comment]
@cmouse want to review?

**Comment by cmouse** [comment]
I'll take a look

File: pdns/pkcs11signers.cc [hunk 1]
```diff
@@ -789,33 +790,31 @@ std::string PKCS11DNSCryptoKeyEngine::hash(const std::string& msg) const {
...
-      SHA384Summer sha;
-      sha.feed(msg);
-      return sha.get();
-    }
-    };
-  };
+
+    unsigned char hash[64];
+    md_init(&md_ctx);
+    md_init_ctx(&md_ctx, md_info);
```
Comment (cmouse) [inline pdns/pkcs11signers.cc]: md_info ends up uninitialized if some weird algorithm is used. Maybe add a default case in select?

**Comment by cmouse** [comment]
Error caused by:
```
Jul 31 21:24:30 Unable to load module '../regression-tests/modules/librandombackend.so': ../regression-tests/modules/librandombackend.so: cannot open shared object file: No such file or directory
Jul 31 21:24:30 DNSBackend unable to load module in random
```

**Comment by Habbie** [comment]
randombackend error is result of another merge to master earlier.

File: pdns/pkcs11signers.cc (hunk 1 above)
Comment (rubenk) [inline pdns/pkcs11signers.cc]: Sure, what should the default case do? Throw a PDNSException?

Comment (cmouse) [inline pdns/pkcs11signers.cc]: Sounds good to me

**Comment by pieterlexis** [comment]
Tests fixed, default case looks like it was added. @cmouse: OK? @rubenk: finished?

**Comment by cmouse** [comment]
LGTM.

**Comment by rubenk** [comment]
@pieterlexis yeah, good to go.

**Comment by mind04** [comment]
Please consider #2688 as a replacement for this pull request. It does use the general md stuff and it add polar 1.3.x and 2.0.x compatibility.

**Comment by rubenk** [comment]
Ooh, even better ;)
Closing this.

**Pull request closed without merging**